src/Data/Auth/Providers/KeycloakAuthContext.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect, useCallback, useMemo } from "react";
+import Keycloak, { KeycloakConfig } from "keycloak-js";
+import { AuthContext } from "./AuthContext";
+
+interface AuthProviderProps {
+  children: React.ReactNode;
+  config: KeycloakConfig;
+}
+
+const KeycloakAuthProvider = ({ children, config }: AuthProviderProps) => {
+  const keycloakInstance = useMemo(() => new Keycloak(config), [config]);
+
+  const getName = useCallback((): string | undefined => {
+    return keycloakInstance &&
+      keycloakInstance.profile &&
+      keycloakInstance.profile.username
+      ? keycloakInstance.profile.username
+      : undefined;
+  }, [keycloakInstance]);
+
+  const [user, setUser] = useState<string | undefined>(undefined);
```

Comment:
Typically, the useStates are placed as first declarations

Reply:
removed that totally, I will verify the implementation though, as I suspect that keycloak could require some tweaks anyway